iavl/iavl_proof2.go

```diff
@@ -58,27 +60,28 @@ func (p *PathToKey) isRightmost() bool {
 	return true
 }
 
-type innerPathPair struct {
-	left  []IAVLProofInnerNode
-	right []IAVLProofInnerNode
+func (p *PathToKey) pop() *PathToKey {
```

Comment:
@cloudhead I have a question about the intention of this method, perhaps a comment above might help? It doesn't look like it mutates the original structure.